graphs/util/db.py

```diff
@@ -1608,6 +1608,298 @@ def find_all_graphs_containing_nodes(uid, search_type, search_word, view_type, d
 
 	return graph_dict.values()
 
+
+def uploadGPMLFile(username, graphJSON, title):
+	'''
+		Uploads GPML file to GraphSpace via /upload.
+		@param username: Owner of graph
+		@param graphJSON: JSON of graph
+		@param title: Title of graph
+	'''
+
+	try:
+		# Create JSON stucture for GraphSpace recognized JSON
```

Comment:
I think we can have a parseGPML method here. This method can parse GPML to graphspace graph object.

We can have graphspace graph class which will define the graph template. This class will allow expose a method to get graphspace graph in JSON format. You do not need to make this class support previous functionality.